### PR TITLE
Nmcli add options

### LINKED ISSUE #2730: nmcli: add ipv4.routing-rules, ipv4.may-fail options and disabled to method6
### Summary

I am trying to configure our network setup which is possible with nmcli command line but there are some missing options in the ansible collections which prevents us from using it. Adding this options will extend the possible configuration that the collection can do and get closer to nmcli command capabilities.

### Issue Type

Feature Idea

### Component Name

nmcli

### Additional Information

This type of configuration is for example used for Lustre and Multi-Rail environments.
https://wiki.lustre.org/LNet_Router_Config_Guide#ARP_flux_issue_for_MR_node
```
nmcli c add type infiniband con-name ib0 \
  ifname ib0 infiniband.mtu 2044 ipv4.method manual \
  ipv4.may-fail false ipv4.addresses 10.0.0.3/16 \
  ipv4.route-metric 0 ipv4.routes "10.0.0.0/16 src=10.0.0.3 table=200" \
  ipv4.routing-rules "priority 0 from 10.0.0.3 lookup 200" \
  ipv6.method disabled connection.zone internal
```

```yaml (paste below)
vars:
  infiniband_interfaces:
    - device: ib0
      type: infiniband
      ip4: 10.0.0.3/16
      mtu: 2044
      route_table: 200
      may_fail4: false

tasks:
  - name: "Configure IB MR interface"
    community.general.nmcli:
       conn_name: "{{ item.device }}"
       ifname: "{{ item.device }}"
       type: "{{ item.type }}"
       ip4: "{{ item.ip4 }}"
       mtu: "{{ item.mtu }}"
       method4: manual
       method6: disabled
       routes4: "{{ item.ip4 | ipaddr('network') }}/{{ item.ip4 | ipaddr('prefix') }} src={{ item.ip4 | ipaddr('address') }} table={{ item.route_table }}"
       route_metric4: 0
       routing_rules4: "priority 0 from {{ item.ip4 | ipaddr('address') }} lookup {{ item.route_table }}"
       may_fail4: false
       state: present

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by omula**
##### SUMMARY

Add option `routing_rules4` and `may_fail4` which translates to `ipv4.routing-rules` and `ipv4.may-fail` respctively.
Add `disabled` option to method6 to be able to completely disabled IPv6.

Fixes #2730 
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/modules/net_tools/nmcli.py

### CODE REVIEW

File: plugins/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -95,6 +95,10 @@
             - Set metric level of ipv4 routes configured on interface.
         type: int
         version_added: 2.0.0
+    routing_rules4:
+        description:
+            - Is the same as in an 'ip route add' command, except always requires specifying a priority.
+        type: str
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:101]: Also for other options.
```suggestion
        type: str
        version_added: 3.2.0
```

File: plugins/modules/net_tools/nmcli.py [hunk 2]
```diff
@@ -126,6 +130,11 @@
         type: str
         choices: [auto, link-local, manual, shared, disabled]
         version_added: 2.2.0
+    may_fail4:
+        description:
+            - If you need I(ip4) configured before network-online.target is reached, set the the property to no.
+        type: bool
+        default: yes
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:137]:
```suggestion
        default: true
```

**Review by aminvakil — COMMENTED** [review]
Thanks for adding this!
Also add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment).

File: changelogs/fragments/2732-nmcli_add_options.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - nmcli - Add option routing_rules4 and may_fail4 (https://github.com/ansible-collections/community.general/issues/2730).
```
Comment (felixfontein) [inline changelogs/fragments/2732-nmcli_add_options.yml:2]:
```suggestion
  - nmcli - add options ``routing_rules4`` and ``may_fail4`` (https://github.com/ansible-collections/community.general/issues/2730).
```

File: changelogs/fragments/2732-nmcli_add_options.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+  - nmcli - Add disabled option to method6 (https://github.com/ansible-collections/community.general/issues/2730).
```
Comment (felixfontein) [inline changelogs/fragments/2732-nmcli_add_options.yml:3]:
```suggestion
  - nmcli - add ``disabled`` value to ``method6`` option (https://github.com/ansible-collections/community.general/issues/2730).
```

File: plugins/modules/net_tools/nmcli.py [hunk 5]
```diff
@@ -165,7 +176,7 @@
             - Configuration method to be used for IPv6
             - If I(ip6) is set, C(ipv6.method) is automatically set to C(manual) and this parameter is not needed.
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:177]:
```suggestion
            - If I(ip6) is set, C(ipv6.method) is automatically set to C(manual) and this parameter is not needed.
            - C(disabled) was added in community.general 3.2.0.
```

**Comment by felixfontein** [comment]
(There are currently CI problems, probably caused by the fastly.com outage.)

**Comment by omula** [comment]
Thanks for all your suggestions. Is there anything pending to fix that I might have missed?

**Comment by felixfontein** [comment]
@omula the unit tests are failing, these have to be fixed.

File: plugins/modules/net_tools/nmcli.py [hunk 6]
```diff
@@ -126,6 +131,12 @@
         type: str
         choices: [auto, link-local, manual, shared, disabled]
         version_added: 2.2.0
+    may_fail4:
+        description:
+            - If you need I(ip4) configured before network-online.target is reached, set the the property to no.
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:136]:
```suggestion
            - If you need I(ip4) configured before C(network-online.target) is reached, set this option to C(false).
```

File: plugins/modules/net_tools/nmcli.py [hunk 7]
```diff
@@ -95,6 +95,11 @@
             - Set metric level of ipv4 routes configured on interface.
         type: int
         version_added: 2.0.0
+    routing_rules4:
+        description:
+            - Is the same as in an 'ip route add' command, except always requires specifying a priority.
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:100]:
```suggestion
            - Is the same as in an C(ip route add) command, except always requires specifying a priority.
```
Can you add an example using this option (and potentially also the others)? I think that would help understanding what this does.

File: changelogs/fragments/2732-nmcli_add_options.yml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - nmcli - add options ``routing_rules4`` and ``may_fail4`` (https://github.com/ansible-collections/community.general/issues/2730).
```
Comment (aminvakil) [inline changelogs/fragments/2732-nmcli_add_options.yml:2]:
```suggestion
  - nmcli - add ``routing_rules4`` and ``may_fail4`` options (https://github.com/ansible-collections/community.general/issues/2730).
```

File: plugins/modules/net_tools/nmcli.py [hunk 9]
```diff
@@ -95,6 +95,11 @@
             - Set metric level of ipv4 routes configured on interface.
         type: int
         version_added: 2.0.0
+    routing_rules4:
+        description:
+            - Is the same as in an C(ip route add) command, except always requires specifying a priority.
+        type: str
+        version_added: 3.2.0
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:102]: Unfortunately 3.2.0 has been released and this PR didn't make it for that.
```suggestion
        version_added: 3.3.0
```

File: plugins/modules/net_tools/nmcli.py [hunk 10]
```diff
@@ -126,6 +131,12 @@
         type: str
         choices: [auto, link-local, manual, shared, disabled]
         version_added: 2.2.0
+    may_fail4:
+        description:
+            - If you need I(ip4) configured before C(network-online.target) is reached, set this option to C(false).
+        type: bool
+        default: true
+        version_added: 3.2.0
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:139]: Here too.
```suggestion
        version_added: 3.3.0
```

File: plugins/modules/net_tools/nmcli.py [hunk 11]
```diff
@@ -164,8 +175,9 @@
         description:
             - Configuration method to be used for IPv6
             - If I(ip6) is set, C(ipv6.method) is automatically set to C(manual) and this parameter is not needed.
+            - C(disabled) was added in community.general 3.2.0.
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:178]: Also here.
```suggestion
            - C(disabled) was added in community.general 3.3.0.
```

**Review by aminvakil — CHANGES_REQUESTED** [review]

**Review by aminvakil — APPROVED** [review]
Thanks for adding these options!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@omula thanks for adding this!
@aminvakil thanks for reviewing!

**Pull request merged**